modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/cxf/clientConfig/SSLContextElytronClientConfigTestCase.java

```diff
@@ -0,0 +1,122 @@
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.xml.namespace.QName;
+import javax.xml.ws.BindingProvider;
+import javax.xml.ws.Service;
+import java.io.File;
+import java.net.URL;
+
+import static org.jboss.wsf.test.JBossWSTest.assertEquals;
+import static org.jboss.wsf.test.JBossWSTest.getResourceURL;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@RunWith(Arquillian.class)
+public class SSLContextElytronClientConfigTestCase {
+
+   private static final String DEPLOYMENT = "sslMutualAuthContextTestCase";
+   private static final String TWO_WAY_SSL_CONTAINER = "ssl-mutual-auth";
+
+   @ArquillianResource
+   private Deployer deployer;
+
+   @ArquillianResource
+   private ContainerController containerController;
+
+
+   @Deployment(name = DEPLOYMENT, testable = false, managed = false)
+   @TargetsContainer(TWO_WAY_SSL_CONTAINER)
+   public static WebArchive createDeployment() {
+      WebArchive archive = ShrinkWrap.create(WebArchive.class, "ssl-mutual-auth.war");
+      archive.setManifest(new StringAsset("Manifest-Version: 1.0\n"
+              + "Dependencies: org.jboss.ws.cxf.jbossws-cxf-client\n"))
+              .addClass(org.jboss.test.ws.jaxws.cxf.httpauth.Hello.class)
+              .addClass(org.jboss.test.ws.jaxws.cxf.httpauth.HelloImpl.class)
+              .addClass(org.jboss.test.ws.jaxws.cxf.httpauth.HelloRequest.class)
+              .addClass(org.jboss.test.ws.jaxws.cxf.httpauth.HelloResponse.class)
+              .addClass(org.jboss.test.ws.jaxws.cxf.httpauth.ObjectFactory.class)
+              .addAsWebInfResource(new File(JBossWSTestHelper.getTestResourcesDir() + "/jaxws/cxf/httpauth/WEB-INF/wsdl/helloMutualSSLport.wsdl"), "wsdl/hello.wsdl");
+      return archive;
+   }
+
+   @Before
+   public void startContainerAndDeploy() throws Exception {
+      if (!containerController.isStarted(TWO_WAY_SSL_CONTAINER)) {
+         containerController.start(TWO_WAY_SSL_CONTAINER);
+         try {
+            deployer.deploy(DEPLOYMENT);
+         } catch (Exception e) {
+            System.err.println(e);
+         }
+      }
+   }
+
+   @Test
+   @RunAsClient
+   public void testConfiguredSSLContext() throws Exception {
+      AuthenticationContext previousAuthContext = AuthenticationContext.getContextManager().getGlobalDefault();
+      try {
+         ElytronClientTestUtils.setElytronClientConfig(JBossWSTestHelper.getTestResourcesDir() + "/jaxws/cxf/clientConfig/META-INF/wildfly-config-ssl-context.xml");
+         QName serviceName = new QName("http://jboss.org/http/security", "HelloService");
+         URL wsdlURL = getResourceURL("jaxws/cxf/httpauth/WEB-INF/wsdl/hello.wsdl");
+         Service service = Service.create(wsdlURL, serviceName);
+         Hello proxy = service.getPort(Hello.class);
+         ((BindingProvider) proxy).getRequestContext().put(BindingProvider.ENDPOINT_ADDRESS_PROPERTY, "https://localhost:13443/ssl-mutual-auth/HelloService");
```

Comment:
could you please parametrize 13443 so that it takes port offset for tested container into account? Eg. https://github.com/jbossws/jbossws-cxf/blob/master/modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/samples/wsse/policy/oasis/WSSecurityPolicyExamples23xTestCase.java#L68-L69

Reply:
Fixed.